### PR TITLE
in case of memcached failure will return same as if item is not cached

### PR DESCRIPTION
**Pull request by jusurb**
in case of failure memcached return false, this will return an empty array as if the tags where not cached at all

### CODE REVIEW

**Comment by lstrojny** [comment]
Can you add an integration test for it?

**Comment by jusurb** [comment]
added integration test, had to refactor a bit \InterNations\Component\Caching\Tests\Zend\Integration\AbstractIntegrationTest to be able to stop start/stop servers from the test

File: tests/InterNations/Component/Caching/Zend/Integration/LibmemcachedIntegrationTest.php [hunk 1]
```diff
@@ -23,4 +27,24 @@ public function setUp()
         );
         $this->backend = new LibmemcachedTaggingBackend($this->memcache);
     }
+
+    public function testReturnEmptyArrayOnConnectionFailure()
+    {
+        $this->backend->save('FooBar', 'id1122', ['tag1', 'tag2', 'tag3']);
+
+        $result = $this->backend->loadTagRevisions(static::STORED_TAGS);
+        $this->assertEquals(Memcached::RES_SUCCESS, $this->backend->getMemcache()->getResultCode());
```
Comment (lstrojny) [inline tests/InterNations/Component/Caching/Zend/Integration/LibmemcachedIntegrationTest.php]: assertSame()

File: tests/InterNations/Component/Caching/Zend/Integration/LibmemcachedIntegrationTest.php [hunk 2]
```diff
@@ -23,4 +27,24 @@ public function setUp()
...
+        $this->assertInternalType('array', $result);
+
+        static::stopAllServers();
+
+        $resultAfterConnectionFailure = $this->backend->loadTagRevisions(static::STORED_TAGS);
+
+        $this->assertEquals(static::MEMCACHED_CONNECTION_FAILURE, $this->backend->getMemcache()->getResultCode());
```
Comment (lstrojny) [inline tests/InterNations/Component/Caching/Zend/Integration/LibmemcachedIntegrationTest.php]: assertSame()

File: tests/InterNations/Component/Caching/Zend/Integration/LibmemcachedIntegrationTest.php [hunk 3]
```diff
@@ -23,4 +27,24 @@ public function setUp()
...
+        $this->assertEquals([], $resultAfterConnectionFailure);
```
Comment (lstrojny) [inline tests/InterNations/Component/Caching/Zend/Integration/LibmemcachedIntegrationTest.php]: assertSame()

**Review by lstrojny — CHANGES_REQUESTED** [review]

**Comment by lstrojny** [comment]
Yippieh!

**Pull request merged**